Jenkinsfile

```diff
@@ -15,38 +22,95 @@ pipeline {
 		timestamps()
 
 		// Prevent concurrent builds
-		disableConcurrentBuilds();
+		disableConcurrentBuilds()
 	}
 	stages {
-		stage ('1.3.3 Build') {
+		stage ('Discover Version Number') {
+			steps {
+				script {
+					def pom = readMavenPom file: 'pom.xml'
+					def currentVersion = pom.getVersion()
+					echo 'Current Version: ' + currentVersion
+
+					VERSION = currentVersion.substring(0, currentVersion.indexOf('-SNAPSHOT'))
+					echo 'Release Version: ' + VERSION
+
+					def parts = VERSION.tokenize('-')
+					echo 'Parts: ' + parts
+
+					def index = parts[1].toInteger()
+					echo 'Index: ' + index
+
+					parts.remove(1)
+					NEW_VERSION = parts.join('-') + '-' + (index + 1)
+					echo 'Next Version: ' + NEW_VERSION
+				}
+			}
+		}
+		stage ('1.3.3 Release Build') {
             when {
-                branch "1.3.x"
+                expression {
+                    return BRANCH_NAME == "1.3.x" && params.RELEASE
+                }
+            }
+            steps {
+                sh "mvn versions:set -B -DnewVersion=$VERSION"
+                sh "mvn -N -B versions:update-child-modules"
+				script {
+					sh "git commit --all --message 'Creating Release $VERSION'"
+					sh "git tag --annotate v$VERSION --message 'Creating Release $VERSION'"
+					sh "git push origin HEAD:${BRANCH_NAME} --tags"
+				}
+                timeout(time: 10, unit: 'MINUTES') {
+                    withMaven(options: [jUnitPublisher(disabled: true)]) {
+	                    sh "mvn -B -V -U -T4 clean deploy -DaltReleaseDeploymentRepository=releases::default::https://nexus.greshamtech.com/content/repositories/thirdparty/"
+                    }
+                }
+            }
+            post {
+                always {
+                       archiveArtifacts caseSensitive: false, onlyIfSuccessful: true, allowEmptyArchive: true, artifacts: 'openid-connect-server-webapp/target/*.war'
```

Comment:
Given you have wrapped the mvn deploy step with `withMaven`, you shouldn't need to manually archive artifacts. Jenkins will make a magic link to Nexus.

Reply:
Just mimicked from what was existing in the Jenkinsfile but get your point, will remove